src/TimingData.cpp

```diff
@@ -755,7 +754,7 @@ bool TimingData::DoesLabelExist( const RString& sLabel ) const
 
 void TimingData::GetBeatAndBPSFromElapsedTime(GetBeatArgs& args) const
 {
-	args.elapsed_time += GAMESTATE->m_SongOptions.GetCurrent().m_fMusicRate * PREFSMAN->m_fGlobalOffsetSeconds;
+	args.elapsed_time += PREFSMAN->m_fGlobalOffsetSeconds;
```

Comment:
Instead of directly adding the global offset to elapsed_time, we should convert the global offset to the sound file offset. Then, we should apply the adjusted offset to `elapsed_time`.